src/engines/adapters/base-adapter.js

```diff
@@ -113,4 +115,39 @@ export default class BaseMediaSourceAdapter {
   destroy() {
     // should do nothing. implemented by the inheritor if necessary.
   }
+
+  selectTrack(track: Track) {
+    if (track) {
+      let success = false;
+      switch (track.type) {
```

Comment:
I don't think we need the _type_ property in Track class.
We will conclude the type by the instance of the track.
For example:
`if (track instanceof AudioTrack){...}`